Code/src/main/java/nl/utwente/group10/haskell/type/Type.java

```diff
@@ -27,6 +30,27 @@ public String toHaskellType() {
     public abstract String toHaskellType(final int fixity);
 
     /**
+     * @param The fixity of the context the type is shown in.
+     * @param The list of types applied to this type.
+     * @return the Haskell (type) representation of this type a list of of types 
+     */
+    public String asTypeAppChain(final int fixity, final ArrayList<Type> args)
```

Comment:
Would prefer to weaken second argument to `List<Type>` instead of `ArrayList<Type>`.